52_wa_diff_shipment.py

```diff
@@ -0,0 +1,263 @@
+    "int64"
+)
+prescriptions_reduced_copy["Population"] = prescriptions_reduced_copy[
+    "Population"
+].astype("int64")
+
+# creating a dataset that has all the drug prescriptions in the state of Florida
+
+washington_prescriptions = prescriptions_reduced_copy[
+    prescriptions_reduced_copy["StateName"] == "Washington"
+]
+# washington_prescriptions_copy = washington_prescriptions.copy()
+washington_prescriptions["shipment_per_100k"] = (
+    (
+        washington_prescriptions["dos_str"]
+        * washington_prescriptions["DOSAGE_UNIT"]
+        * washington_prescriptions["MME_Conversion_Factor"]
+    )
+    / (washington_prescriptions["Population"])
+    * 100000
+)
+
+washington_prescriptions
+
+# calculating and displaying total number of drug prescriptions that took place in washington. results are grouped and displayed year and county wise
+
+washington_prescriptions_result = (
+    washington_prescriptions.groupby(["Year", "county_test"])["shipment_per_100k"]
+    .sum()
+    .reset_index()
+)
+# washington_prescriptions_result = washington_prescriptions.groupby(["Year"])["shipment_per_100k"].sum().reset_index()
+
+washington_prescriptions_result.head()
+
+## Diff-in-Diff
+
+other_states_shipment = prescriptions_reduced_copy[
+    (prescriptions_reduced_copy["StateName"] != "Washington")
+]
+# other_states_shipment = prescriptions_reduced_copy[(prescriptions_reduced_copy["StateName"] == "Alabama")|(prescriptions_reduced_copy["StateName"] == "Georgia")|(prescriptions_reduced_copy["StateName"] == "Mississippi")|(prescriptions_reduced_copy["StateName"] == "South Carolina")|(prescriptions_reduced_copy["StateName"] == "Tennessee")]
+
+other_states_shipment
+
+other_states_shipment_copy = other_states_shipment.copy()
+
+# other_states_shipment_copy["shipment_per_100k"] = (other_states_shipment_copy["QUANTITY"] / other_states_shipment_copy["Population"]) * 100_000
+other_states_shipment_copy["shipment_per_100k"] = (
+    (
+        other_states_shipment_copy["dos_str"]
+        * other_states_shipment_copy["DOSAGE_UNIT"]
+        * other_states_shipment_copy["MME_Conversion_Factor"]
+    )
+    / (other_states_shipment_copy["Population"])
+    * 100000
+)
+
+other_states_shipment_copy
+
+# calculating and displaying results for drug related deaths - year, state and county wise
+
+other_states_result = (
+    other_states_shipment_copy.groupby(["Year", "StateName", "county_test"])[
+        "shipment_per_100k"
+    ]
+    .sum()
+    .reset_index()
+)
+other_states_result.head()
+
+washington_summary = pd.DataFrame(
+    washington_prescriptions_result.describe()["shipment_per_100k"]
+).rename(columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Texas"})
+other_states_summary = pd.DataFrame(
+    other_states_result.describe()["shipment_per_100k"]
+).rename(
+    columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Comp States"}
+)
+stats = pd.concat([washington_summary, other_states_summary], axis=1)
```

Comment:
For better readability of code, include spaces between calculations for washington_stats, other_states_stats and stats.

Reply:
Thanks! I'll do notebook formatting next time uploading the notebook!